util-test/src/iosMain/kotlin/care/data4life/sdk/util/test/CommonResourceLoader.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2021 D4L data4life gGmbH / All rights reserved.
+ *
+ * D4L owns all legal rights, title and interest in and to the Software Development Kit ("SDK"),
+ * including any intellectual property rights that subsist in the SDK.
+ *
+ * The SDK and its documentation may be accessed and used for viewing/review purposes only.
+ * Any usage of the SDK for other purposes, including usage for the development of
+ * applications/third-party applications shall require the conclusion of a license agreement
+ * between you and D4L.
+ *
+ * If you are interested in licensing the SDK for your own applications/third-party
+ * applications and/or if you’d like to contribute to the development of the SDK, please
+ * contact D4L by email to help@data4life.care.
+ */
+
+package care.data4life.sdk.util.test
+
+import care.data4life.sdk.util.NSDataMapper
+import care.data4life.sdk.util.test.lang.FileNotFoundError
+import kotlinx.cinterop.memScoped
+import platform.Foundation.NSData
+import platform.Foundation.NSFileManager
+import platform.Foundation.NSString
+import platform.Foundation.NSUTF8StringEncoding
+import platform.Foundation.create
+import platform.posix.F_OK
+import platform.posix.access
+
+actual class CommonResourceLoader actual constructor(
+    projectDir: AbsolutePath
+) {
+    private val projectPath = projectDir
+
+    actual fun exists(path: Path, root: Path?): Boolean {
+        return access(
+            CommonPathResolver.resolvePath(
+                projectPath,
+                root,
+                path
+            ),
+            F_OK
+        ) == 0
+    }
+
+    private fun readBytes(path: String): NSData {
+        return memScoped {
+            NSFileManager.defaultManager().contentsAtPath(
+                path
+            ) ?: RuntimeException("Out of Memory")
+        } as NSData
+    }
+
+    @Throws(FileNotFoundError::class)
+    actual fun load(path: Path, root: Path?): String {
```

Comment:
is this method actually needed? if we are returning. a string, we might need the string encoding as a parameter (e.g nsAttributedString needs utf16)

Reply:
Could you maybe provide a example, where you think that might break?

Reply:
And yes...the method itself is already in use.

Reply:
done